internal/namespaces/namespaces.go

```diff
@@ -9,8 +9,16 @@ type Namespace struct {
 	Name string `json:"name"`
 }
 
-func ListNamespaces(k kubernetes.Interface) []Namespace {
-	namespaceList, _ := k.CoreV1().Namespaces().List(metav1.ListOptions{})
+type NamespaceService interface {
```

Comment:
I need to do some more reading so I don't just blindly borrow Spring-isms and graft them onto go. Due to my ignorance, I'm starting with this naming convention.